### PR TITLE
DEV: Remove use of Discourse application code in migration.

### PR DESCRIPTION
**Pull request by tgxworld**
This makes the migration less brittle to changes in the future.

Follow-up to https://github.com/tgxworld/discourse/commit/21a356e3af34f15234d09cbe0e4bdb3eff408f5d

### CODE REVIEW

**Comment by tgxworld** [comment]
@martin-brennan Can you review this PR? Thank you :bow:

File: db/post_migrate/20220214224506_reset_custom_emoji_post_bakes_version_secure_fix.rb [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 class ResetCustomEmojiPostBakesVersionSecureFix < ActiveRecord::Migration[6.1]
   def up
-    if SiteSetting.secure_media
+    secure_media_enabled = execute("SELECT value FROM site_settings WHERE name = 'secure_media'")
```
Comment (martin-brennan) [inline db/post_migrate/20220214224506_reset_custom_emoji_post_bakes_version_secure_fix.rb:5]: Hmm can't we just use `DB.query_single` and avoid this ntuples wrangling?

**Review by martin-brennan — COMMENTED** [review]
Argh this rule gets me so much, didn't even think of `SiteSetting` being part of it. Just a change suggested

File: db/post_migrate/20220214224506_reset_custom_emoji_post_bakes_version_secure_fix.rb (hunk 1 above)
Comment (martin-brennan) [reply db/post_migrate/20220214224506_reset_custom_emoji_post_bakes_version_secure_fix.rb:5]: This is used in plenty of other post_migrate code

Comment (tgxworld) [reply db/post_migrate/20220214224506_reset_custom_emoji_post_bakes_version_secure_fix.rb]: Good point :) I've updated the PR to reflect that.

**Review by martin-brennan — APPROVED** [review]

**Comment by martin-brennan** [comment]
Nice, thanks!

**Pull request merged**